### PR TITLE
feat(sca): add Image Referencer for CloudFormation

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- added Image Referencer capabilities to CloudFormation resources. I kept the implementation very similar to the Terraform one even it was not needed to add some of the classes, because we don't have multiple providers there. But I think it is easier to understand and maintain, if they are kept similar 🙂 
- needed to adjust the Terraform Batch resource a bit, because the node properties are not supported in Terraform

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/cloudformation/image_referencer/test_runner_aws_resources.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    assert (
+        sca_image_report.image_cached_results[0]["dockerImageName"]
+        == "public.ecr.aws/aws-containers/hello-app-runner:latest"
+    )
+    assert (
+        "cloudformation/image_referencer/resources/aws/apprunner.yaml:AWS::AppRunner::Service.AppRunner"
+        in sca_image_report.image_cached_results[0]["relatedResourceId"]
+    )
+    assert sca_image_report.image_cached_results[0]["packages"] == [
+        {"type": "os", "name": "zlib", "version": "1.2.12-r1", "licenses": ["Zlib"]}
```
Comment (ChanochShayner) [inline tests/cloudformation/image_referencer/test_runner_aws_resources.py:64]: 💯

**Review by ChanochShayner — APPROVED** [review]
👑

File: checkov/cloudformation/image_referencer/provider/aws.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    return image_names
+
+
+# needs to be at the bottom to add the defined functions
+SUPPORTED_AWS_IMAGE_RESOURCE_TYPES = {
+    "AWS::AppRunner::Service": extract_images_from_aws_apprunner_service,
+    "AWS::Batch::JobDefinition": extract_images_from_aws_batch_job_definition,
+    "AWS::CodeBuild::Project": extract_images_from_aws_codebuild_project,
+    "AWS::ECS::TaskDefinition": extract_images_from_aws_ecs_task_definition,
+    "AWS::Lightsail::Container": extract_images_from_aws_lightsail_container_service_deployment_version,
```
Comment (schosterbarak) [inline checkov/cloudformation/image_referencer/provider/aws.py:97]: what about:
EKS, Lambda, ECR, Copilot?

Comment (gruebel) [reply checkov/cloudformation/image_referencer/provider/aws.py:97]: EKS: you can only create cluster, node groups and some other management stuff, nothing related to containers. those are added via other means like normal Kubernetes manifests or helm
Lambda: only allows privately stored images in ECR, which can't scan anyway
ECR: only defines the registry and its settings, but you can' reference any images in the resources
Copilot: is a CLI tool it has its own configuration files

**Review by schosterbarak — APPROVED** [review]

**Pull request merged**